benchmarks/ShuffleBench.cu

```diff
@@ -25,10 +27,12 @@ static void benchmarkFunction( benchmark::State& state )
 
     ContainerType in_container( num_to_shuffle );
     ContainerType out_container( num_to_shuffle );
-
+    int seed=0;
     for( auto _ : state )
     {
-        shuffler( in_container, out_container );
+        shuffler( in_container, out_container, seed);
+        checkCudaError(cudaDeviceSynchronize());
```

Comment:
This should be handled by the implementation not the benchmark. This is redundant.

Reply:
Unnecessary synchronisation in implementation should always be avoided. 

It's necessary purely for benchmarking. Accurately timing cuda code because host code proceeds asynchronously, CPU based timers think it has finished earlier than it has.

Reply:
My understanding was thrust implicitly synchronizes, hence why I did it that the implementation is responsible. Otherwise we should add a method shuffler.sync() that can synchronize a specific stream etc. so the sync is not more heavyweight than necessary